tests/xfstests/run.pm

```diff
@@ -379,31 +294,62 @@ END_CMD
     enter_cmd("$cmd");
 }
 
-sub reload_loop_device {
-    my $self = shift;
-    assert_script_run("losetup -fP $INST_DIR/test_dev");
-    my $scratch_amount = script_output("ls $INST_DIR/scratch_dev* | wc -l");
-    my $scratch_num    = 1;
-    while ($scratch_amount >= $scratch_num) {
-        assert_script_run("losetup -fP $INST_DIR/scratch_dev$scratch_num", 300);
-        $scratch_num += 1;
-    }
-    script_run('losetup -a');
-    format_partition("$INST_DIR/test_dev", $FSTYPE);
-}
+# Run a single test and write log to file
+# test - test to run(e.g. xfs/001)
+sub test_run {
+    my ($self,     $test) = @_;
+    my ($category, $num)  = split(/\//, $test);
+    eval {
+        $test_start = time();
+        assert_script_run("timeout " . ($TIMEOUT - 5) . " $TEST_WRAPPER '$test' -k | tee $LOG_DIR/$category/$num", $TIMEOUT);
+        $test_duration = time() - $test_start;
+    };
+    if ($@) {
+        $status        = 'FAILED';
+        $test_duration = time() - $test_start;
+        sleep(2);
+        copy_log($category, $num, 'out.bad');
+        copy_log($category, $num, 'full');
+        copy_log($category, $num, 'dmesg');
+        copy_fsxops($category, $num);
+        collect_fs_status($category, $num);
+        if (get_var('BTRFS_DUMP', 0) && (check_var 'XFSTESTS', 'btrfs')) { dump_btrfs_img($category, $num); }
+
+        #return to VNC due to grub2 and linux-login needle
+        select_console('root-console');
+        power_action('reboot', keepconsole => 1);
+        reconnect_mgmt_console if is_pvm;
+        $self->wait_boot(bootloader_time => 500);
 
-# Umount TEST_DEV and SCRATCH_DEV
-sub umount_xfstests_dev {
-    script_run('umount ' . get_var('XFSTESTS_TEST_DEV') . ' &> /dev/null')    if get_var('XFSTESTS_TEST_DEV');
-    script_run('umount ' . get_var('XFSTESTS_SCRATCH_DEV') . ' &> /dev/null') if get_var('XFSTESTS_SCRATCH_DEV');
-    if (get_var('XFSTESTS_SCRATCH_DEV_POOL')) {
-        script_run("umount $_ &> /dev/null") foreach (split ' ', get_var('XFSTESTS_SCRATCH_DEV_POOL'));
+        $self->select_serial_terminal;
+        # Save kdump data to KDUMP_DIR if not set "NO_KDUMP"
+        unless (get_var('NO_KDUMP')) {
+            unless (save_kdump($test, $KDUMP_DIR, vmcore => 1, kernel => 1, debug => 1)) {
+                # If no kdump data found, write warning to log
+                my $msg = "Warning: $test crashed SUT but has no kdump data";
+                script_run("echo '$msg' >> $LOG_DIR/$category/$num");
+            }
+        }
+
+        # Add test status to STATUS_LOG file
+        log_add($test, $status, $test_duration);
+
+        # Reload loop device after a reboot
+        if (get_var('XFSTESTS_LOOP_DEVICE')) {
+            reload_loop_device;
+        }
```

Comment:
```suggestion
        reload_loop_device if get_var('XFSTESTS_LOOP_DEVICE');
        }
```

if you like :)

Reply:
no, even if you would write it right ...

Reply:
yeah, right ;P